lib/r10k/module_loader/puppetfile.rb

```diff
@@ -74,6 +82,30 @@ def load
         raise R10K::Error.wrap(e, _("Failed to evaluate %{path}") % {path: @puppetfile_path})
       end
 
+      def load_metadata
+        with_readable_puppetfile(@puppetfile_path) do
+          self.load_metadata!
+        end
+      end
+
+      def load_metadata!
+        dsl = R10K::ModuleLoader::Puppetfile::DSL.new(self, metadata_only: true)
+        dsl.instance_eval(puppetfile_content(@puppetfile), @puppetfile)
+
+        @existing_module_versions_by_name = @existing_module_metadata.map {|mod| [ mod.name, mod.version ] }.to_h
+        empty_load_output.merge(modules: @existing_module_metadata)
+
+      rescue SyntaxError, LoadError, ArgumentError, NameError => e
+        logger.warn "Unable to preload Puppetfile"
```

Comment:
The other version of this includes a little more info about why things failed. I know we don't want to rethrow here, but do you think it'd be valuable to include at least the message from the original exception?

Reply:
Updated